src/jit/regset.cpp

```diff
@@ -3458,19 +3458,52 @@ bool                genIsProperRegPair(regPairNo regPair)
  *
  *  Given a register that is an argument register 
  *   returns the next argument register 
+ *
+ *  Note: that this method will return a non arg register 
+ *   when given REG_ARG_LAST
+ *
```

Comment:
I'm curious why you chose this behavior instead of returning REG_NA or something.